altapay.php

```diff
@@ -0,0 +1,2353 @@
+                    'message' => "Reservation captured successfully"
+                )
+            );
+            die();
+        } // REFUND
+        elseif ($action == 'Refund') {
+            try {
+                $refundAmount = Tools::getValue('amount');
+                if (Tools::getValue('goodwillrefund') == 'yes') {
+                    $goodWillRefund = true;
+                }
+                $finalOrderLines = $this->populateOrderLinesFromPost($orderLines, $orderLineGiftWrap, $orderID, $goodWillRefund);
+
+                // Add a dummy orderLine array in case no orderLines are parsed in the refund
+                if ($finalOrderLines === array() && $goodWillRefund) {
+                    $finalOrderLines = $this->createDummyOrderLinesArr($refundAmount);
+                }
+                $api->refundAmount($paymentID, $finalOrderLines, $refundAmount);
+                $refundUpdate = markAsRefund($paymentID, $this->getItemCaptureRefundQuantityCount($finalOrderLines));
+                if (!$refundUpdate) {
+                    throw new Exception("The refund could not be updated in database");
+                }
+            } catch (Exception $e) {
+                $response = json_decode($e->getMessage(), true);
+                saveLastErrorMessage($paymentID, $response['responseMsg']);
+
+                echo json_encode(
+                    array(
+                        'status' => $response['responseResult'],
+                        'message' => "Could not refund payment. " . $response['responseMsg']
+                    )
+                );
+                die();
+            }
+
+            echo json_encode(
+                array(
+                    'status' => "success",
+                    'message' => "Payment refunded successfully"
+                )
+            );
+            die();
+        } // RELEASE
+        elseif ($action == 'Release') {
+            try {
+                $api->release($paymentID, $action);
+                updatePaymentStatus($paymentID, "Payment Released");
+            } catch (Exception $e) {
+                $response = json_decode($e->getMessage(), true);
+                saveLastErrorMessage($paymentID, $response['responseMsg']);
+
+                echo json_encode(
+                    array(
+                        'status' => $response['responseResult'],
+                        'message' => "Could not release reservation. " . $response['responseResult'] . ": " . $response['responseMsg']
+                    )
+                );
+                die();
+            }
+            echo json_encode(
+                array(
+                    'status' => "success",
+                    'message' => "Reservation released successfully"
+                )
+            );
+            die();
+        }
+    }
+
+    /* ******************************** */
+
+    /**
+     * Method for generating order lines from order backend
+     * @param $orderLines
```

Comment:
`@param` with out a type are invalid
```suggestion
     * @param int[] $orderLines
```